### ISSUE TITLE
[bug] use-memo false positive in shorthand arrow function syntax

### ISSUE THREAD

**Issue by burtek** [Rel1cx/eslint-react#1611]
### Describe the bug

When using `() => useMemo()` syntax, the rule reports issue about unused `useMemo` output.
This does not happen when using `() => { return useMemo(); }` syntax.

### Reproduction

```ts
const use1 = () => useMemo(() => 1, []);
```

With `@eslint-react/use-memo` set to `"error"`

Currently reports:
> The return value of 'useMemo' must be assigned to a variable. Calling 'useMemo' without capturing its return value is likely a mistake — use 'useEffect' for side effects instead

### Expected behavior

no error

### Platform and versions

```Text
node v24.10.0
@eslint-react/eslint-plugin 3.0.0-rc.1
```

### Stack trace

```Text

```
